lib/collection.js

```diff
@@ -1990,7 +1994,7 @@ Collection.prototype.parallelCollectionScan = deprecate(function(options, callba
   options.readPreference = resolveReadPreference(this, options);
 
   // Add a promiseLibrary
-  options.promiseLibrary = this.s.promiseLibrary;
```

Comment:
I'm a little confused about what setting `options.promiseLibrary` before running `executeLegacyOperation` does. As you can see [here](https://github.com/mongodb/node-mongodb-native/pull/2340/files#diff-50e3aa130a4f97a42ee2cf111c7b1d9dR377), `executeLegacyOperation` ignores the `options` and grabs the `promiseLibrary` from the `topology`. If I'm not missing something, we can just delete anything that sets `options.promiseLibrary` before a call to `executeLegacyOperation`.